### PR TITLE
replace hasAcls() with mailboxHasRights()

### PR DESCRIPTION
**Pull request by hamza221**
Part of [#7057](https://github.com/nextcloud/mail/issues/7057)

### CODE REVIEW

File: src/tests/unit/components/Envelope.vue.spec.js [hunk 1]
```diff
@@ -80,7 +81,7 @@ describe('Envelope', () => {
 			localVue,
 		})
 
-		expect(view.vm.hasSeenAcl).toBe(true)
+		expect(mailboxHasRights(view.vm.mailbox,'s')).toBe(true)
```
Comment (ChristophWurst) [inline src/tests/unit/components/Envelope.vue.spec.js:84]: if the component no longer has a `hasSeenAcl` drop this test

if there is a `hasSeenAcl` that calls `mailboxHasRights` keep the test and test `hasSeenAcl` like before

File: src/components/NavigationMailbox.vue [hunk 2]
```diff
@@ -379,7 +379,7 @@ export default {
 			return this.$route.params.mailboxId === this.mailbox.databaseId
 		},
 		isValidDropTarget() {
-			if (this.isActive || !this.hasInsertAcl) {
+			if (this.isActive || !mailboxHasRights(this.mailbox, 'i')) {
```
Comment (kesselb) [inline src/components/NavigationMailbox.vue:382]: I think hasInsertAcl would improve the readability here.

Comment (hamza221) [reply src/components/NavigationMailbox.vue:382]: on it

**Review by ChristophWurst — APPROVED** [review]
Clever code reuse and still readable. I like it :shipit:

**Pull request merged**